python/pyarrow/_gcsfs.pyx

```diff
@@ -136,6 +140,8 @@ cdef class GcsFileSystem(FileSystem):
         if retry_time_limit is not None:
             time_limit_seconds = retry_time_limit.total_seconds()
             options.retry_limit_seconds = time_limit_seconds
+        if project_id is not None:
+            options.project_id = tobytes(project_id)
```

Comment:
```suggestion
            options.project_id = <c_string>tobytes(project_id)
```

Might help? (didn't test it)

Reply:
Thanks that was good a pointer and solved the issue I was having!